app/js/sagas/orderSaga.js

```diff
@@ -0,0 +1,82 @@
+import {
+  call,
+  take,
+  takeEvery,
+  fork,
+  put,
+  cancel,
+  select,
+} from 'redux-saga/effects';
+import { format, addMinutes } from 'date-fns';
+import encounterRest from '../rest/encounterRest';
+import { getDateRange } from '../utils/helpers';
+import { DISCONTINUE_ORDER } from '../actions/actionTypes';
+import fetchLabOrders from '../actions/labOrders/fetchLabOrders';
+
+
+// eslint-disable-next-line consistent-return
+function* getMatchingEncounter(order) {
+  const state = yield select();
+  const patientUuid = order.patient;
+  const encounterTypeUuid = order.encounterType;
+  const locationUuid = order.location.uuid;
+  const provider = state.encounterRoleReducer.encounterRole.uuid;
+  const duration = format(addMinutes(new Date(), -60));
+
+  try {
+    const encounterResponse = yield call(
+      encounterRest.fetchEncountersByPatient,
+      patientUuid,
+      encounterTypeUuid,
+      provider,
+      locationUuid,
```

Comment:
It would be better if we could pass in the date here as well, to restrict the size of the returned respond, but if we currently don't have an endpoint that supports dates, I'd be fine with this for now.  The main thing would be to see if we can reduce it not to include the full representation of the encounter... for a long-time patient, they might have 50+ encounters with the same provider and this would be a huge response payload.

Reply:
@mogoodrich tried that but it doesn't seem to work, it still returns all the results. maybe i'm not doing it right

Reply:
I can take a look...